src/triage/component/postmodeling/contrast/evaluator.py

```diff
@@ -0,0 +1,543 @@
+        self.train_matrix_uuid = model_metadata.loc[0, 'train_matrix_uuid']
+        self.pred_matrix_uuid = model_metadata.loc[0, 'matrix_uuid']
+        self.train_matrix = None
+        self.pred_matrix = None
+        self.preds = None
+        self.feature_importances = None
+        self.model_metrics = None
+
+    def __str__(self):
+        s = 'Model object for model_id: {}'.format(self.model_id)
+        s += '\n Model Group: {}'.format(self.model_group_id)
+        s += '\n Model type: {}'.format(self.model_type)
+        s += '\n Model hyperparameters: {}'.format(self.hyperparameters)
+        s += '\Matrix hashes (train,test): {}'.format([self.train_matrix_uuid,
+                                                      self.pred_matrix_uuid])
+        return s
+
+    def _load_predictions(self):
+        preds = pd.read_sql("""
+                SELECT model_id,
+                       entity_id,
+                       as_of_date,
+                       score,
+                       label_value,
+                       COALESCE(rank_abs, RANK() OVER(ORDER BY score DESC)) AS rank_abs,
+                       rank_pct,
+                       test_label_timespan
+                FROM test_results.predictions
+                WHERE model_id = {}
+                AND label_value IS NOT NULL
+                """.format(self.model_id),
+                con=db_engine)
+        self.preds = preds
+
+    def _load_feature_importances(self):
+        features = pd.read_sql("""
+               SELECT model_id,
+                      feature,
+                      feature_importance,
+                      rank_abs
+               FROM train_results.feature_importances
+               WHERE model_id = {}
+               """.format(self.model_id),
+               con=db_engine)
+        self.feature_importances = features
+
+    def _load_metrics(self):
+        model_metrics = pd.read_sql("""
+                SELECT model_id,
+                       metric,
+                       metric_parameter,
+                       value,
+                       num_labeled_examples,
+                       num_labeled_above_treshold,
+                       num_positive_labels
+                FROM test_results.evaluation
+                WHERE model_id = {}
+                """.format(self.model_id),
+                con=db_engine)
+        self.model_metrics = model_metrics
+
+    def _fetch_matrix(self, matrix_hash, path):
+        '''
+        Load matrices into object (Beware!)
+        This can be a memory intensive process. This function will use the
+        stored model parameters to load matrices from a .csv file. If the path
+        is a s3 path, the method will use s3fs to open it. If the path is a local
+        machine path, it will be opened as a pandas dataframe.
+
+        Arguments:
+            - PATH <s3 path or local path>
+        '''
+        if 's3' in path:
+            mat = download_s3(str(path + self.train_matrix_uuid))
```

Comment:
`pandas` can read from `s3` with `read_csv`

Reply:
Didn't know that! I'll try it and use the `boto` to read the credentials.